add tests via molecule

ltalirz opened the pull request:
fixes #26 

This PR
 * adds test configuration via molecule
 * runs tests on Ubuntu 16.04 and 18.04 via Travis CI (as well as Debian 9 & 10)
 * adds yamllint

Furthermore it currently runs the role without any parameters and only tests that the role runs and is idempotent.
One could of course add further tests later on.

ltalirz commented:
Note: the build [currently fails](https://travis-ci.org/ltalirz/ansible-dokku/builds/631690739) with 
```
    RUNNING HANDLER [dokku_bot.ansible_dokku : start dokku-daemon] *****************
    fatal: [instance]: FAILED! => {"changed": false, "msg": "Could not find the requested service dokku-daemon: host"}
```
Perhaps you can point me in the right direction?

Note that the idempotency test will anyhow fail until #35 is merged.

josegonzalez commented on .travis.yml:
```diff
@@ -0,0 +1,39 @@
+  include:
+    - name: Ubuntu-18.04
+      env:
+        - DOCKER_IMAGE: marvelnccr/ubuntu-docker-base
```
Can you include the Dockerfile for this image? Or maybe we start a new dokku repo for managing that base image.

josegonzalez commented:
For the failing test, maybe the `install` make target isn't loading the new service properly. See [here](https://github.com/dokku/dokku-daemon/blob/master/Makefile#L14) for how it does it, maybe you can spot the error?

ltalirz commented:
> For the failing test, maybe the install make target isn't loading the new service properly. See here for how it does it, maybe you can spot the error?

Indeed - this failed because my docker image didn't have systemd. I've now switched to the docker images by geerlingguy, which specifically install it (see e.g. [here](https://github.com/geerlingguy/docker-ubuntu1804-ansible/blob/master/Dockerfile)).
Since these images are very well maintained, I don't think it's necessary to have a clone just for dokku. If you prefer I can switch to specific versions for each OS instead of using `latest` but I would probably even go with `latest`.

The tests on Ubuntu 18.04 and 16.04 pass now - I've also added tests for Debian 9 and 10, which should be supported according to the [README](https://github.com/dokku/ansible-dokku#platform-requirements).
Those tests currently fail but I believe this can be fixed in a separate PR.


Note: The little fix that I introduced in the test playbook (was part of my docker images) was needed to get around the following [error during the installation of `herokuish`](https://travis-ci.org/dokku/ansible-dokku/builds/631921557#L565):
```
Importing herokuish into docker (around 5 minutes)", "Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"

josegonzalez commented:
@ltalirz Thanks for the great work here! Mind going through and fixing debian tests? I'll get a release out once thats done :)
